core/offchain-worker-executor/src/executor.rs

```diff
@@ -184,10 +184,34 @@ impl<
 		Ok(())
 	}
 
-	fn send_parentchain_effects(&self, parentchain_effects: Vec<OpaqueCall>) -> Result<()> {
-		let extrinsics = self
-			.extrinsics_factory
-			.create_extrinsics(parentchain_effects.as_slice(), None)?;
+	fn send_parentchain_effects(&self, parentchain_effects: Vec<ParentchainCall>) -> Result<()> {
+		let integritee_calls: Vec<OpaqueCall> = parentchain_effects
+			.iter()
+			.filter_map(|parentchain_call| parentchain_call.as_integritee())
+			.collect();
+		let target_a_calls: Vec<OpaqueCall> = parentchain_effects
+			.iter()
+			.filter_map(|parentchain_call| parentchain_call.as_target_a())
+			.collect();
+		let target_b_calls: Vec<OpaqueCall> = parentchain_effects
+			.iter()
+			.filter_map(|parentchain_call| parentchain_call.as_target_b())
+			.collect();
```

Comment:
This might be a performance bottleneck at some point, `collect` is O(n). How about a:

```rust

struct ParentchainEffects: {
     integritee: Vec<OpaqueCall>,
     target_a: Vec<OpaqueCall>,
     target_b: Vec<OpaqueVall>,
}
```

Reply:
However, we could also change the signatures of `create_extrinsics` to something with `impl Iterator` and then we can omit the collect.

Reply:
#1510

Reply:
#1510